src/engine/GroupBy.cpp

```diff
@@ -857,7 +857,10 @@ GroupBy::isSupportedAggregate(sparqlExpression::SparqlExpression* expr) {
   if (expr->isDistinct()) return std::nullopt;
 
   // `expr` is not a nested aggregated
-  if (expr->children().front()->containsAggregate()) return std::nullopt;
+  if (!expr->children().empty() &&
+      expr->children().front()->containsAggregate()) {
+    return std::nullopt;
+  }
```

Comment:
```suggestion
  if (std::ranges::any_of(expr->children(), [](const auto& ptr) {
        return ptr->containsAggregate();
      })) {
    return std::nullopt;
  }
```

Alternatively this variant does have fewer implicit assumptions about the span of children.

Reply:
I like the alternative, it is more idiomatic and the Coverage tool will like it better.
The support for COUNT(*) is missing in the HashMap optimization, as the optimization was implemented before COUNT(*), and I forgot to add this.
Maybe we'll do this some time in the future, but at least now we avoid the segfault...